packages/material-ui/src/styles/withStyles.d.ts

```diff
@@ -55,6 +55,8 @@ export type WithStyles<
 
 export interface StyledComponentProps<ClassKey extends string = string> {
   classes?: Partial<ClassNameMap<ClassKey>>;
+  className?: string;
+  style?: React.CSSProperties;
```

Comment:
Are those actually necessary? As far as I know `style` is simply spread to the root component. Only `className` is passed to the `classnames` package (and could technically be an object but I'm not sure this is always possible so better leave `className`).

Reply:
Good catch, I'll revert that.